### PR TITLE
listen on specified addresses

### LINKED ISSUE #18554: ShardingSphere-Proxy listen on specified addresses
### Is your feature request related to a problem?

For security.

### Describe the feature you would like.

Support listening on specific addresses like the MySQL or PostgreSQL.

Netty bind: https://stackoverflow.com/questions/30957207/netty-listen-on-several-addresses-ports-with-a-single-serverbootstrap

### PR DESCRIPTION
**Pull request by csonezp**
Fixes #18554 .

To be compatible with both ways, I modified the start script and gave default values for port and conf path .
If the user does not set port or conf path, they will be set to -1/${DEPLOY_DIR}/conf.
The default value for port is -1, that mean user did not specify a port, and it will return Optional.empty() when getPort.

@TeslaCN is this solution correct? can you give me some suggestions to improve the code quality? Thanks!

Changes proposed in this pull request:
- support listen on specified addresses

### CODE REVIEW

File: shardingsphere-distribution/shardingsphere-proxy-distribution/src/main/resources/bin/start.sh [hunk 1]
```diff
@@ -61,10 +61,23 @@ JAVA_MEM_OPTS=" -server -Xmx2g -Xms2g -Xmn1g -Xss1m -XX:AutoBoxCacheMax=4096 -XX
...
+
 print_usage() {
-    echo "usage: start.sh [port] [config_dir]"
+    echo "usage:"
+    echo "start.sh [port] [config_dir]"
     echo "  port: proxy listen port, default is 3307"
-    echo "  config_dir: proxy config directory, default is conf"
+    echo "  config_dir: proxy config directory, default is 'conf'"
+    echo ""
+    echo "start.sh [-h addresses] [-p port] [-c /path/to/conf]"
```
Comment (TeslaCN) [inline shardingsphere-distribution/shardingsphere-proxy-distribution/src/main/resources/bin/start.sh:74]: We have already changed the `-h`.

Comment (csonezp) [reply shardingsphere-distribution/shardingsphere-proxy-distribution/src/main/resources/bin/start.sh]: > We have already changed the `-h`.

done

**Review by TeslaCN — CHANGES_REQUESTED** [review]
Hi @csonezp 
I tried your branch and I found a problem:

I wrote an absolute path of config directory:
```bash
bin/start.sh 33306 /home/wuweijie/dev/shardingsphere/proxy/master/mysql/conf
```

But the absolute path was prefixed with `DEPLOY_DIR`
```
$ tail logs/stdout.log                                                                                                                                                                                                                                                                                             [16:44:09]
        at org.apache.shardingsphere.proxy.backend.config.ProxyConfigurationLoader.load(ProxyConfigurationLoader.java:60)
        at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:49)
Exception in thread "main" java.io.FileNotFoundException: /home/wuweijie/projects/shardingsphere/shardingsphere-distribution/shardingsphere-proxy-distribution/target/apache-shardingsphere-5.1.3-SNAPSHOT-shardingsphere-proxy-bin/home/wuweijie/dev/shardingsphere/proxy/master/mysql/conf/server.yaml (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:216)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at org.apache.shardingsphere.infra.yaml.engine.YamlEngine.unmarshal(YamlEngine.java:54)
        at org.apache.shardingsphere.proxy.backend.config.ProxyConfigurationLoader.loadServerConfiguration(ProxyConfigurationLoader.java:74)
        at org.apache.shardingsphere.proxy.backend.config.ProxyConfigurationLoader.load(ProxyConfigurationLoader.java:60)
        at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:49)
```

**Comment by csonezp** [comment]
> Hi @csonezp I tried your branch and I found a problem:
> 
> I wrote an absolute path of config directory:
> 
> ```shell
> bin/start.sh 33306 /home/wuweijie/dev/shardingsphere/proxy/master/mysql/conf
> ```
> 
> But the absolute path was prefixed with `DEPLOY_DIR`
> 
> ```
> $ tail logs/stdout.log                                                                                                                                                                                                                                                                                             [16:44:09]
>         at org.apache.shardingsphere.proxy.backend.config.ProxyConfigurationLoader.load(ProxyConfigurationLoader.java:60)
>         at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:49)
> Exception in thread "main" java.io.FileNotFoundException: /home/wuweijie/projects/shardingsphere/shardingsphere-distribution/shardingsphere-proxy-distribution/target/apache-shardingsphere-5.1.3-SNAPSHOT-shardingsphere-proxy-bin/home/wuweijie/dev/shardingsphere/proxy/master/mysql/conf/server.yaml (No such file or directory)
>         at java.base/java.io.FileInputStream.open0(Native Method)
>         at java.base/java.io.FileInputStream.open(FileInputStream.java:216)
>         at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
>         at org.apache.shardingsphere.infra.yaml.engine.YamlEngine.unmarshal(YamlEngine.java:54)
>         at org.apache.shardingsphere.proxy.backend.config.ProxyConfigurationLoader.loadServerConfiguration(ProxyConfigurationLoader.java:74)
>         at org.apache.shardingsphere.proxy.backend.config.ProxyConfigurationLoader.load(ProxyConfigurationLoader.java:60)
>         at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:49)
> ```

@TeslaCN my bad, I will go through the script and write a test report. Is there a way to test the script file in the source code?

And I found a bug in master branch: if user wrote an absolute path, the script  will merge  the absolute path and ${DEPLOY_DIR},  and then add the result to ${CLASS_PATH}. Like this:
![image](https://user-images.githubusercontent.com/3190994/177762773-1bbfd115-68bb-428a-b6b3-818fa817e392.png)
should i fix it?

**Comment by TeslaCN** [comment]
You could fix it if you like.

**Comment by csonezp** [comment]
@TeslaCN

**Review by TeslaCN — APPROVED** [review]
Hi @csonezp
Thank you.

**Comment by tristaZero** [comment]
Hello @csonezp Thanks for this nice feature, which will be included in the next release, look forward!
@TeslaCN Patient review!

**Pull request merged**